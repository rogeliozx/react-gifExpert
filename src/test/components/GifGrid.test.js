import React from 'react';
import GifGrid from './../../components/GifGrid';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { useFetchGifs } from './../../hooks/useFetchGifs';
jest.mock('./../../hooks/useFetchGifs');

describe('Test <GifGrid/>', () => {
  useFetchGifs.mockReturnValue({
    data: [],
    loading: true,
  });
  const cetegory = 'One Punch';
  const wrapper = shallow(<GifGrid category={cetegory} />);
  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should show items when load images useFetchGifs', () => {
    const cetegory = 'One Punch';
    const gifs = [
      {
        id: 'adfadsf',
        url: 'test.com',
        title: 'test',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={cetegory} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
