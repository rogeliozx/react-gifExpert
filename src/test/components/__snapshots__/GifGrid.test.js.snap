// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid/> should render 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className=" animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test <GifGrid/> should show items when load images useFetchGifs 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="adfadsf"
      key="adfadsf"
      title="test"
      url="test.com"
    />
  </div>
</Fragment>
`;
